lib/grpc/client/adapters/mint/connection_process/connection_process.ex

```diff
@@ -0,0 +1,177 @@
+defmodule GRPC.Client.Adapters.Mint.ConnectionProcess do
+  use GenServer
+
+  alias GRPC.Client.Adapters.Mint.ConnectionProcess.State
+  alias GRPC.Client.Adapters.Mint.StreamResponseProcess
+
+  require Logger
+
+  def start_link(scheme, host, port, opts \\ []) do
+    GenServer.start_link(__MODULE__, {scheme, host, port, opts})
+  end
+
+  def request(pid, method, path, headers, body, opts \\ []) do
+    GenServer.call(pid, {:request, method, path, headers, body, opts})
+  end
+
+  def disconnect(pid) do
+    GenServer.call(pid, {:disconnect, :brutal})
+  end
+
+  def stream_request_body(pid, request_ref, body) do
+    GenServer.call(pid, {:stream_body, request_ref, body})
+  end
+
+  ## Callbacks
+
+  @impl true
+  def init({scheme, host, port, opts}) do
+    # The current behavior in gun is return error if the connection wasn't successful
+    # Should we do the same here?
+    case Mint.HTTP.connect(scheme, host, port, opts) do
+      {:ok, conn} ->
+        {:ok, State.new(conn)}
+
+      {:error, reason} ->
+        # TODO check what's better: add to state map if connection is alive?
+        # TODO Or simply stop the process and handle the error on caller?
```

Comment:
Since we have a process per connection, I think killing it might represent better what's happening